### PR TITLE
Don't give the illusion that we can version hashes

### PR DESCRIPTION
**Pull request by edsko**

### CODE REVIEW

File: ouroboros-consensus/src/Ouroboros/Consensus/Network/NodeToNode.hs [hunk 1]
```diff
@@ -241,6 +242,12 @@ defaultCodecs ccfg version = Codecs {
     dec :: SerialiseNodeToNode blk a => forall s. Decoder s a
     dec = decodeNodeToNode ccfg version
 
+    encHash :: HeaderHash blk -> Encoding
+    encHash = Serialise.encode . toRawHash (Proxy @blk)
+
+    decHash :: forall s. Decoder s (HeaderHash blk)
+    decHash = fromRawHash (Proxy @blk) <$> Serialise.decode
```
Comment (mrBliss) [inline ouroboros-consensus/src/Ouroboros/Consensus/Network/NodeToNode.hs:249]: It's the third time I see these now, we should define them near `ConvertRawHash`

File: ouroboros-consensus-byron-test/src/Ouroboros/Consensus/ByronDual/Node/Serialisation.hs [hunk 2]
```diff
@@ -99,11 +99,6 @@ instance DecodeDisk DualByronBlock (AnnTip DualByronBlock) where
 
 instance SerialiseNodeToNodeConstraints DualByronBlock
 
--- | @'HeaderHash' 'DualByronBlock'@
-instance SerialiseNodeToNode DualByronBlock ByronHash where
-  encodeNodeToNode _ _ = encodeByronHeaderHash
-  decodeNodeToNode _ _ = decodeByronHeaderHash
```
Comment (mrBliss) [inline ouroboros-consensus-byron-test/src/Ouroboros/Consensus/ByronDual/Node/Serialisation.hs:105]: Lets' get rid of `encodeByronHeaderHash` and `decodeByronHeaderHash` as well.

File: ouroboros-consensus/src/Ouroboros/Consensus/Network/NodeToNode.hs (hunk 1 above)
Comment (edsko) [reply ouroboros-consensus/src/Ouroboros/Consensus/Network/NodeToNode.hs:249]: Ah, okay, will do.

File: ouroboros-consensus-byron-test/src/Ouroboros/Consensus/ByronDual/Node/Serialisation.hs (hunk 2 above)
Comment (edsko) [reply ouroboros-consensus-byron-test/src/Ouroboros/Consensus/ByronDual/Node/Serialisation.hs:105]: These are still used internally.

**Comment by edsko** [comment]
bors merge

**Pull request merged**